src/ffi_ptr_ext.rs

```diff
@@ -0,0 +1,28 @@
+use crate::{ffi, instance::Py2, PyAny, PyResult, Python};
+
+mod ffi_ptr_ext_sealed {
+    use super::*;
+
+    pub trait Sealed {}
+
+    impl Sealed for *mut ffi::PyObject {}
+}
+
+use ffi_ptr_ext_sealed::Sealed;
+
+pub(crate) trait FfiPtrExt: Sealed {
+    unsafe fn assume_owned_or_ffi_error(self, py: Python<'_>) -> PyResult<Py2<'_, PyAny>>;
```

Comment:
I think I would prefer the naming to match what it forwards to, i.e. `assume_owned_or_err` and `assume_owned`.

Reply:
Sure thing, I don't really love the `owner_or_err` names but I guess at this point they are more or less entrenched, let's keep things uniform for now and if I want to pull at that thread another day, I'll do so 😂